plugins/scripts/inject-config.js

```diff
@@ -2,6 +2,24 @@
 const fs = require('fs');
 const path = require('path');
 
+function createJsonWithDefaultValues(optimoveCredentialsParam, optimoveMobileCredentialsParam ,optimoveInAppConsentStrategy) {
+    var optimoveCredentials ="";
+    var optimoveMobileCredentials = "";
+    
+    if (optimoveCredentialsParam) {
```

Comment:
Worth using isEmpty here? What values can the args have?

Reply:
Perhaps we can use some tertiary condition when building the returned object, for example

```
return {
  OPTIMOVE_CREDS: isEmpty(optimoveCredentials) ? "" : optimoveCredentials,
...}
```

What do you think?

(i.e. can rename function params to drop `Param` suffix and lose the local vars)

Reply:
I actually think it's better that way because it covers more cases, for example the case which the args are undefined, and it also covers the case that the string is empty or null, what do you think? (the args will behave according to the optimove.json, if we put a key without value it will be null or empty and if don't put a key it's undefined)

Reply:
I think isEmpty should handle null/empty string/undefined values too.

I guess truthiness works with some coercion in JS, so it's roughly equivalent. I'm maybe thinking how to simplify the structure of variables + parameters, but maybe it's not a big deal.

One further consideration might be testing the values is a string before passing it through. What do you think?

Reply:
sure sounds like an important thing to check., in the situation where inside the optimove.json there won't be a key for the property (for ex inAppConsentStrategy), the value will be undefined, should I cover undefined values also or we can just assume that the json will be in a right format?